test/controllers/renewal/HowCashPaymentsReceivedControllerSpec.scala

```diff
@@ -0,0 +1,110 @@
+import services.RenewalService
+import uk.gov.hmrc.http.cache.client.CacheMap
+import utils.{AmlsSpec, AuthorisedFixture}
+
+import scala.concurrent.Future
+
+class HowCashPaymentsReceivedControllerSpec extends AmlsSpec {
+
+  lazy val mockDataCacheConnector = mock[DataCacheConnector]
+  lazy val mockRenewalService = mock[RenewalService]
+
+  val receiveCashPayments = CashPayments(CashPaymentsCustomerNotMet(true), Some(HowCashPaymentsReceived(PaymentMethods(true,true,Some("other")))))
+  val doNotreceiveCashPayments = CashPayments(CashPaymentsCustomerNotMet(false), None)
+
+  trait Fixture extends AuthorisedFixture {
+    self => val request = addToken(authRequest)
+
+    val controller = new HowCashPaymentsReceivedController (
+      dataCacheConnector = mockDataCacheConnector,
+      authConnector = self.authConnector,
+      renewalService = mockRenewalService
+    )
+
+    when(mockRenewalService.getRenewal(any(),any(),any()))
+      .thenReturn(Future.successful(None))
+
+    when(mockRenewalService.updateRenewal(any())(any(),any(),any()))
+      .thenReturn(Future.successful(new CacheMap("", Map.empty)))
+  }
+
+  "HowCashPaymentsReceived" when {
+    "get is called" must {
+      "load the page" when {
+        "renewal data is found for how cash payments received and pre-populate the data" in new Fixture {
+          when(mockRenewalService.getRenewal(any(),any(),any()))
+            .thenReturn(Future.successful(Some(Renewal(receiveCashPayments = Some(receiveCashPayments)))))
+
+          val result = controller.get()(request)
+          status(result) mustEqual OK
+
+          val page = Jsoup.parse(contentAsString(result))
+
+          page.select("input[type=checkbox][name=cashPaymentMethods.courier][value=true]").hasAttr("checked") must be(true)
+          page.select("input[type=checkbox][name=cashPaymentMethods.direct][value=true]").hasAttr("checked") must be(true)
+          page.select("input[type=checkbox][name=cashPaymentMethods.other][value=true]").hasAttr("checked") must be(true)
+          page.select("input[type=text][name=cashPaymentMethods.details]").first().`val`() must be("other")
+        }
+
+        "no renewal data is found and show an empty form" in new Fixture {
+          val result = controller.get()(request)
+          status(result) mustEqual OK
+
+          val page = Jsoup.parse(contentAsString(result))
+          page.select("input[type=checkbox][name=cashPaymentMethods.courier][value=true]").hasAttr("checked") must be(false)
+          page.select("input[type=checkbox][name=cashPaymentMethods.direct][value=true]").hasAttr("checked") must be(false)
+          page.select("input[type=checkbox][name=cashPaymentMethods.other][value=true]").hasAttr("checked") must be(false)
+          page.select("input[type=text][name=cashPaymentMethods.details]").first().`val`() must be("")
+        }
+      }
+    }
+
+    "post is called" must {
+      "show a bad request with an invalid request" in new Fixture {
```

Comment:
Consider rewording such that its reads a little better? 

`post is called` when {
`an invalid request is made` must {
   `show a bad request` in {...}
}

Reply:
sure I'll change that